version.go

```diff
@@ -38,7 +38,8 @@ func clientnegotiate(ctx context.Context, ch Channel, version string) (string, e
 			return "", fmt.Errorf("unsupported server version: %v", version)
 		}
 
-		if int(v.MSize) > ch.MSize() {
+		if int(v.MSize) < ch.MSize() {
```

Comment:
The client already sent their size. This scales down the local msize if the server responds with a smaller value. The protocol works like this:

1. Client sends msize.
2. Server responds with msize.
  v.Msize > msize, use server's msize.

At least, this was how I understand from study of 9p. Do you have documentation countering this? What does plan9 do?